test/verify/check-realms

```diff
@@ -610,6 +615,40 @@ for s in key p12 pem; do docker cp freeipa:jane.$s .; done
             m.execute("systemctl unmask sssd-ifp")
 
 
+@skipImage("adcli not on test images", "debian-stable", "debian-testing", "ubuntu-stable", "ubuntu-2004")
+class TestAD(TestRealms, CommonTests):
+    def setUp(self):
+        super().setUp()
+        self.admin_user = "Administrator"
+        self.admin_password = "foobarFoo123"
+        self.expected_server_software = "active-directory"
+        self.machines['services'].execute("/run-samba-domain")
+        # Wait for AD to come up and DNS to work as expected
+        wait(lambda: self.machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
+        # DNS is not sufficient yet, needs to start LDAP server as well
+        self.machine.execute("until nc -z f0.cockpit.lan 389; do sleep 1; done")
+        # also wait for Kerberos
+        self.machine.execute("set -e; until echo %s | kinit %s@COCKPIT.LAN; do sleep 1; done; kdestroy" % (self.admin_password, self.admin_user))
+
+        # allow sudo access to domain admins; FIXME: Is there a server-side setting for this,
+        # similar to "ipa-advise enable-admins-sudo"?
+        self.machine.write("/etc/sudoers.d/domain-admins", r"%domain\ admins@COCKPIT.LAN ALL=(ALL) ALL")
+
+    def checkBackendSpecifics(self):
+        '''Check domain backend specific integration'''
+
+        pass
+
+    def checkBackendSpecificCleanup(self):
+        '''Check domain backend specific integration after leaving domain'''
+
+        pass
```

Comment:
these two are not going to be implemented at all, or is this just tmp state?

Reply:
I'm not sure yet. It would certainly be nice to add some of the integration that we do with IPA to AD as well, like getting a centrally managed SSL certificate for Cockpit. I haven't yet looked whether this is possible. For now I'd just like to keep this as a future extension point, and of course the method needs to be defined for the shared test to work at all.